File: app/src/main/java/hearsilent/universalcollapsingtoolbarlayouttablayoutexample/fragment/DemoFragment.java
package hearsilent.universalcollapsingtoolbarlayouttablayoutexample.fragment;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import hearsilent.universalcollapsingtoolbarlayouttablayoutexample.R;

public class DemoFragment extends Fragment {

	View mRootView;
	RecyclerView mRecyclerView;

	Context mContext;

	DemoAdapter mAdapter;

	List<String> mList = new ArrayList<>();

	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
	                         @Nullable Bundle savedInstanceState) {
		mRootView = inflater.inflate(R.layout.fragment_demo, container, false);

		initValues();
		findViews();
		setUpViews();

		return mRootView;
	}

	@Override
	public void onAttach(@NonNull Context context) {
		super.onAttach(context);
		mContext = context;
	}

	private void initValues() {
		for (int i = 0; i < 100; i++) {
			String item = "HearSilent " + i;
			mList.add(item);
		}
	}

	private void findViews() {
		mRecyclerView = mRootView.findViewById(R.id.recyclerView);
	}

	private void setUpViews() {
		mAdapter = new DemoAdapter();
		mRecyclerView.setHasFixedSize(true);
		mRecyclerView.setItemAnimator(new DefaultItemAnimator());
		mRecyclerView.setLayoutManager(new LinearLayoutManager(mContext));
		mRecyclerView.setAdapter(mAdapter);
	}

	public class DemoAdapter extends RecyclerView.Adapter<DemoAdapter.DemoViewHolder> {

		public class DemoViewHolder extends RecyclerView.ViewHolder {

			public TextView textView;

			public DemoViewHolder(View view) {
				super(view);

				textView = view.findViewById(R.id.textView);
			}
		}

		@NonNull
		@Override
		public DemoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
			View view = LayoutInflater.from(parent.getContext())
					.inflate(R.layout.list_demo, parent, false);
			return new DemoViewHolder(view);
		}

		@Override
		public void onBindViewHolder(DemoViewHolder holder, int position) {
			holder.textView.setText(mList.get(position));
		}

		@Override
		public int getItemCount() {
			return mList.size();
		}
	}

}

File: app/src/main/java/hearsilent/universalcollapsingtoolbarlayouttablayoutexample/fragment/DemoFragment.kt
package hearsilent.universalcollapsingtoolbarlayouttablayoutexample.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import hearsilent.universalcollapsingtoolbarlayouttablayoutexample.R
import hearsilent.universalcollapsingtoolbarlayouttablayoutexample.databinding.FragmentDemoBinding
import hearsilent.universalcollapsingtoolbarlayouttablayoutexample.databinding.ListDemoBinding
import hearsilent.universalcollapsingtoolbarlayouttablayoutexample.fragment.DemoFragment.DemoAdapter.DemoViewHolder

class DemoFragment : Fragment() {

    private lateinit var mBinding: FragmentDemoBinding

    private lateinit var mAdapter: DemoAdapter
    private val mList = ArrayList<String>()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        mBinding = FragmentDemoBinding.inflate(inflater, container, false)

        initValues()
        setUpViews()

        return mBinding.root
    }

    private fun initValues() {
        for (i in 0..99) {
            val item = "HearSilent $i"
            mList.add(item)
        }
    }

    private fun setUpViews() {
        mAdapter = DemoAdapter()
        mBinding.recyclerView.setHasFixedSize(true)
        mBinding.recyclerView.setItemAnimator(DefaultItemAnimator())
        mBinding.recyclerView.setLayoutManager(LinearLayoutManager(requireContext()))
        mBinding.recyclerView.setAdapter(mAdapter)
    }

    inner class DemoAdapter : RecyclerView.Adapter<DemoViewHolder>() {

        inner class DemoViewHolder(view: View) : RecyclerView.ViewHolder(view) {
            var binding = ListDemoBinding.bind(view)
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DemoViewHolder {
            val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_demo, parent, false)
            return DemoViewHolder(view)
        }

        override fun onBindViewHolder(holder: DemoViewHolder, position: Int) {
            holder.binding.textView.text = mList[position]
        }

        override fun getItemCount(): Int {
            return mList.size
        }
    }
}
